src/spikeinterface/postprocessing/amplitude_scalings.py

```diff
@@ -0,0 +1,304 @@
+        Whether to load precomputed spike amplitudes, if they already exist.
+    peak_sign: str
+        The sign to compute maximum channel:
+            - 'neg'
+            - 'pos'
+            - 'both'
+    return_scaled: bool
+        If True and recording has gain_to_uV/offset_to_uV properties, amplitudes are converted to uV.
+    outputs: str
+        How the output should be returned:
+            - 'concatenated'
+            - 'by_unit'
+    {}
+
+    Returns
+    -------
+    amplitude_scalings: np.array or list of dict
+        The amplitude scalings.
+            - If 'concatenated' all amplitudes for all spikes and all units are concatenated
+            - If 'by_unit', amplitudes are returned as a list (for segments) of dictionaries (for units)
+    """
+    if load_if_exists and waveform_extractor.is_extension(AmplitudeScalingsCalculator.extension_name):
+        sac = waveform_extractor.load_extension(AmplitudeScalingsCalculator.extension_name)
+    else:
+        sac = AmplitudeScalingsCalculator(waveform_extractor)
+        sac.set_params(sparsity=sparsity, max_dense_channels=max_dense_channels, ms_before=ms_before, ms_after=ms_after)
+        sac.run(**job_kwargs)
+
+    amps = sac.get_data(outputs=outputs)
+    return amps
+
+
+compute_amplitude_scalings.__doc__.format(_shared_job_kwargs_doc)
+
+
+def _init_worker_amplitude_scalings(
+    recording,
+    spikes,
+    all_templates,
+    unit_ids,
+    unit_ids_to_channel_indices,
+    nbefore,
+    nafter,
+    cut_out_before,
+    cut_out_after,
+    return_scaled,
+):
+    # create a local dict per worker
+    worker_ctx = {}
+    worker_ctx["recording"] = recording
+    worker_ctx["spikes"] = spikes
+    worker_ctx["all_templates"] = all_templates
+    worker_ctx["nbefore"] = nbefore
+    worker_ctx["nafter"] = nafter
+    worker_ctx["cut_out_before"] = cut_out_before
+    worker_ctx["cut_out_after"] = cut_out_after
+    worker_ctx["margin"] = max(nbefore, nafter)
+    worker_ctx["return_scaled"] = return_scaled
+
+    # construct handy unit_inds -> channel_inds
+    worker_ctx["unit_inds_to_channel_indices"] = {
+        unit_ind: unit_ids_to_channel_indices[unit_id] for unit_ind, unit_id in enumerate(unit_ids)
+    }
+
+    return worker_ctx
+
+
+def _amplitude_scalings_chunk(segment_index, start_frame, end_frame, worker_ctx):
+    # recover variables of the worker
+    spikes = worker_ctx["spikes"]
+    recording = worker_ctx["recording"]
+    all_templates = worker_ctx["all_templates"]
+    unit_inds_to_channel_indices = worker_ctx["unit_inds_to_channel_indices"]
+    nbefore = worker_ctx["nbefore"]
+    cut_out_before = worker_ctx["cut_out_before"]
+    cut_out_after = worker_ctx["cut_out_after"]
+    margin = worker_ctx["margin"]
+    return_scaled = worker_ctx["return_scaled"]
+
+    i0 = np.searchsorted(spikes["segment_ind"], segment_index)
+    i1 = np.searchsorted(spikes["segment_ind"], segment_index + 1)
+    spikes_in_segment = spikes[i0:i1]
```

Comment:
The segment slicing could be done in worker init once.

Reply:
But the workers could span multiple segments no?

Reply:
yes but the first serahsorted can be done once for all.